contracts/mocks/TokenProxy.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+// Original contract: https://github.com/convex-eth/platform/blob/main/contracts/contracts/wrappers/ConvexStakingWrapper.sol
+pragma solidity 0.8.6;
+import "@yield-protocol/utils-v2/contracts/token/ERC20.sol";
+
+contract TokenProxy {
+    ERC20 immutable _curve;
+
+    constructor(ERC20 curve_) {
+        _curve = curve_;
+    }
+
+    /// @param _owner The address from which the balance will be retrieved
+    /// @return balance the balance
+    function balanceOf(address _owner) external view returns (uint256 balance) {
+        return _curve.balanceOf(_owner);
+    }
+
+    /// @notice send `_value` token to `_to` from `msg.sender`
+    /// @param _to The address of the recipient
+    /// @param _value The amount of token to be transferred
+    /// @return success Whether the transfer was successful or not
+    function transfer(address _to, uint256 _value) external returns (bool success) {
+        return _curve.transfer(_to, _value);
```

Comment:
This won't work, because `msg.sender` to `TokenProxy` is the user, and `msg.sender` to `_curve` is `TokenProxy`.

`return _curve.transferFrom(msg.sender, _to, _value)` might be what you need.

Reply:
I am trying to mock the `crv` & `cvx` contract in the `ConvexStakingWrapper` at the addresses mentioned in the [contract](https://github.com/yieldprotocol/vault-v2/blob/fix/convex-wrapper-mocking/contracts/utils/convex/ConvexStakingWrapper.sol#L37-L38).


I got this working by minting the tokens to `TokenProxy` [here](https://github.com/yieldprotocol/vault-v2/blob/fix/convex-wrapper-mocking/test/092_convex_wrapper.ts#L236-L237).

So, in theory the token is being transferred from `TokenProxy` & not the `ConvexPoolMock` which is the user calling the `transfer` but it is ok as we are just trying to simulate the rewards received from the `ConvexPool` as seen [here](https://github.com/yieldprotocol/vault-v2/blob/fix/convex-wrapper-mocking/contracts/utils/convex/ConvexStakingWrapper.sol#L283). So, it doesn't matter for the tests as to who actually is sending the rewards token as long as it is received after the `getReward` is called.

Reply:
All clear, thank you!